websites/management/commands/unpublish_sites.py

```diff
@@ -0,0 +1,125 @@
+"""Unpublish course webpages provided to the command that have no dependencies. If there are dependencies, print the dependencies."""
+import json
+
+from django.conf import settings
+from django.db.models import Q
+
+from content_sync.api import trigger_unpublished_removal
+from main.management.commands.filter import WebsiteFilterCommand
+from users.models import User
+from websites.constants import (
+    CONTENT_TYPE_COURSE_LIST,
+    CONTENT_TYPE_PAGE,
+    CONTENT_TYPE_RESOURCE_COLLECTION,
+    PUBLISH_STATUS_NOT_STARTED,
+)
+from websites.models import Website, WebsiteContent
+from websites.serializers import WebsiteBasicSerializer, WebsiteContentSerializer
+
+
+class Command(WebsiteFilterCommand):
+    """Unpublish course webpages provided to the command that have no dependencies. If there are dependencies, print the dependencies."""
+
+    help = __doc__
+
+    def add_arguments(self, parser):
+        super().add_arguments(parser)
+        parser.add_argument(
+            "--course-ids",
+            nargs="+",
+            dest="course_ids",
+            help="List of course webpages to be unpublished",
+        )
+
+        parser.add_argument(
+            "--user",
+            dest="user",
+            help="Username (email address) for unpublishing courses",
+        )
+
+    def check_email(self, email_id):
+        try:
+            user = User.objects.get(email__exact=email_id)
+            return user
+        except User.DoesNotExist:
+            return None
+
+    def handle(self, *args, **options):
+        super().handle(*args, **options)
+        if not options["course_ids"]:
+            self.stdout.write("Please provide a list of course ids to unpublish.")
+        elif not options["user"]:
+            self.stdout.write("Please provide a username to unpublish courses.")
+        elif not self.check_email(options["user"]):
```

Comment:
Won't be necessary if `required=True` is added to `user` arg in `add_arguments` above.

Reply:
Updated in 9b2e6ed52d822fba85c459bcc7e67aeb81be9400.